vocabularies/Common.xml

```diff
@@ -1450,6 +1450,68 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
         <Annotation Term="Core.Description" String="Base URL for WebSocket connections" />
       </Term>
 
+      <TypeDefinition Name="NamespaceQualifiedActionName" UnderlyingType="Edm.String">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="The name of an action in scope, qualified with a namespace (not an alias)" />
+      </TypeDefinition>
+
+      <Term Name="Recommendations" Type="Common.RecommendationsType" Nullable="false" AppliesTo="EntitySet Singleton Collection">
+        <Annotation Term="Core.AppliesViaContainer" />
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="AI-based recommendations for entities in the annotation target" />
+      </Term>
+      <ComplexType Name="RecommendationsType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="Action" Type="Common.NamespaceQualifiedActionName">
+          <Annotation Term="Core.Description" String="Name of an action (or function) that retrieves recommendations (see [template](#Template_RecommendationsAction))" />
+        </Property>
+        <Property Name="Dependencies" Type="Collection(Common.RecommendationsDependency)">
+          <Annotation Term="Core.Description" String="Predictors for each recommendation target" />
+        </Property>
+      </ComplexType>
+      <Action Name="Template_RecommendationsAction" IsBound="true">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Template for actions that retrieve AI-based recommendations for the bound entity and its related entities" />
+        <Annotation Term="Core.LongDescription">
+          <String>The template action itself cannot be invoked.</String>
+        </Annotation>
+        <Parameter Name="Entity" Type="Edm.EntityType" Nullable="false">
+          <Annotation Term="Core.Description" String="Entity relative to which the `EntityKeys` are interpreted" />
+        </Parameter>
+        <Parameter Name="EntityKeys" Type="Edm.EntityType" Nullable="false">
```

Comment:
What's the purpose of this parameter? The action is instance-bound, so how do these key values relate to the key values of the binding parameter?

Reply:
`EntityKeys` can contain _nested_ entity keys_, for example
```
{"SalesOrder": "A", "_Items": [{
  "SalesOrderItem": "010"
}, {
  "SalesOrderItem": "020"
}]]
```
if only recommendations for header and items 010 and 020 are desired.

Reply:
Please add this to the description.

Reply:
And can the "root" key differ from the key of the binding instance?